core/src/main/java/org/elasticsearch/index/query/HasChildQueryBuilder.java

```diff
@@ -84,34 +157,82 @@ public HasChildQueryBuilder setShortCircuitCutoff(int shortCircuitCutoff) {
     /**
      * Sets inner hit definition in the scope of this query and reusing the defined type and query.
      */
-    public HasChildQueryBuilder innerHit(QueryInnerHitBuilder innerHit) {
-        this.innerHit = innerHit;
+    public HasChildQueryBuilder innerHit(QueryInnerHits queryInnerHits) {
+        this.queryInnerHits = queryInnerHits;
         return this;
     }
 
+    /**
+     * Returns inner hit definition in the scope of this query and reusing the defined type and query.
+     */
+    public QueryInnerHits innerHit() {
+        return queryInnerHits;
+    }
+
+    /**
+     * Returns the children query to execute.
+     */
+    public QueryBuilder query() {
+        return query;
+    }
+
+    /**
+     * Returns the child type
+     */
+    public String childType() {
+        return type;
+    }
+
+    /**
+     * Returns how the scores from the matching child documents are mapped into the parent document.
+     */
+    public ScoreType scoreType() {
+        return scoreType;
+    }
+
+    /**
+     * Returns the minimum number of children that are required to match for the parent to be considered a match.
+     * The default is {@value #DEFAULT_MAX_CHILDREN}
+     */
+    public int minChildren() {
+        return minChildren;
+    }
+
+    /**
+     * Returns the maximum number of children that are required to match for the parent to be considered a match.
+     * The default is {@value #DEFAULT_MIN_CHILDREN}
+     */
+    public int maxChildren() { return maxChildren; }
+
+    /**
+     * Returns what cut off point only to evaluate parent documents that contain the matching parent id terms
+     * instead of evaluating all parent docs. The default is {@value #DEFAULT_SHORT_CIRCUIT_CUTOFF}
+     */
+    public int shortCircuitCutoff() {
+        return shortCircuitCutoff;
+    }
+
     @Override
     protected void doXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(NAME);
         builder.field("query");
-        queryBuilder.toXContent(builder, params);
-        builder.field("child_type", childType);
-        if (scoreType != null) {
-            builder.field("score_type", scoreType);
+        query.toXContent(builder, params);
+        builder.field("child_type", type);
+        if (scoreType != ScoreType.NONE) {
+            builder.field("score_type", scoreType.name().toLowerCase(Locale.ROOT));
         }
-        if (minChildren != null) {
+        if (minChildren != DEFAULT_MIN_CHILDREN) {
             builder.field("min_children", minChildren);
         }
-        if (maxChildren != null) {
+        if (maxChildren != DEFAULT_MAX_CHILDREN) {
             builder.field("max_children", maxChildren);
         }
-        if (shortCircuitCutoff != null) {
+        if (shortCircuitCutoff != DEFAULT_SHORT_CIRCUIT_CUTOFF) {
             builder.field("short_circuit_cutoff", shortCircuitCutoff);
         }
```

Comment:
I think we can simplify here and print everything out, default values included, that's what we went for in all of the other queries too.